core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -1374,6 +1377,10 @@ public RexLiteral makeLiteral(boolean b) {
     return b ? booleanTrue : booleanFalse;
   }
 
+  public RexLiteral makeUuidLiteral(@Nullable UUID uuid) {
+    return new RexLiteral(uuid, typeFactory.createSqlType(SqlTypeName.UUID), SqlTypeName.UUID);
```

Comment:
Can the UUID be null? According to SqlUuidLiteral‘s implementation, it can't.
```
  public UUID getUuid() {
    return (UUID) requireNonNull(value, "value");
  }
```

Reply:
SqlUuidLiteral cannot be null, but the RexLiteral can be null, due to simplifications of expressions such as `CAST(NULL AS UUID)`.

Reply:
BTW: this pattern holds true for many other literals, such as `SqlTimestampLiteral`